pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -105,80 +105,50 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
-	}
-
-	// delete non-default with proper annotation if not publishingStratagy CR
-	err = r.deleteIngressWithAnnotation(instance.Spec.ApplicationIngress, ingressControllerList)
-	if err != nil {
-		log.Error(err, "Cannot delete ingresscontroller with annotation")
-		return reconcile.Result{}, err
-	}
+	// Loop through ingress controllers defined in PublishingStrategy spec
+	for _, ingressDefinition := range instance.Spec.ApplicationIngress {
 
-	// create list of applicationIngress
-	var ingressNotOnCluster []cloudingressv1alpha1.ApplicationIngress
-
-	exisitingIngressMap := convertIngressControllerToMap(ingressControllerList.Items)
-
-	// loop through every applicationingress in publishing strategy
-	for _, publishingStrategyIngress := range instance.Spec.ApplicationIngress {
-		if !checkExistingIngress(exisitingIngressMap, &publishingStrategyIngress) {
-			ingressNotOnCluster = append(ingressNotOnCluster, publishingStrategyIngress)
+		// Set the namespaced name
+		namespacedName := types.NamespacedName{Name: getIngressName(ingressDefinition.DNSName), Namespace: ingressControllerNamespace}
+		// Default
+		if ingressDefinition.Default {
+			namespacedName = types.NamespacedName{Name: "default", Namespace: ingressControllerNamespace}
 		}
-	}
-
-	// ----------------------------TODO: remove these debug logs afterwards --------------------------------------
-	var serializedIngressControllerList bytes.Buffer
-	_ = serializer.Encode(ingressControllerList, &serializedIngressControllerList)
-	log.Info(fmt.Sprintf("initial ingresscontroller list: %s", serializedIngressControllerList.String()))
-	log.Info(fmt.Sprintf("appingress on publishingstrategy CR: %+v", instance.Spec.ApplicationIngress))
-	log.Info(fmt.Sprintf("ingress to reconcile: %+v", ingressNotOnCluster))
-	// -----------------------------------------------------------------------------------------------------------
 
-	for _, appingress := range ingressNotOnCluster {
-		newCertificate := &corev1.LocalObjectReference{
-			Name: appingress.Certificate.Name,
+		desiredIngressController := generateIngressController(ingressDefinition)
+		// Try to get matching IngressController
+		ingressController := &operatorv1.IngressController{}
+		err = r.client.Get(context.TODO(), namespacedName, ingressController)
+		if err != nil {
+			// Attempt to create the CR if not found
+			if k8serr.IsNotFound(err) {
+				err = r.client.Create(context.TODO(), desiredIngressController)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+				// If the CR was created, requeue PublishingStrategy
+				return reconcile.Result{Requeue: true}, nil
+			}
+			return reconcile.Result{}, err
 		}
 
-		if appingress.Default == true {
-			err := r.defaultIngressHandle(appingress, ingressControllerList, newCertificate)
+		// If the IngressController exists, ensure it matches PublishingStrategies definition
+		if !reflect.DeepEqual(ingressController.Spec, desiredIngressController.Spec) {
```

Comment:
There are certain fields like `Spec.Domain` that the installer leaves blank, which allows them to be the default (you can see what the default is in `Status.Domain`.

The other thing we have to do if we are doing this is enumerate *all* the spec fields to ensure we are setting them appropriately. There are some fields we currently don't manage/compare with the existing logic, and a shift to using DeepEqual, while ultimately will fix this once and for all, could break functionality on the non-compared fields if we clobber them.

Reply:
@cblecker The default CR gets a spec check in this workflow currently, whic covers the first part of what you're talking about.